packages/shared/src/domain/patient/patient.ts

```diff
@@ -1,13 +1,19 @@
 import { z } from "zod";
+import { createNonEmptryStringSchema } from "../../common/string";
+import { demographicsSchema } from "./demographics";
 
-export const patientCreateResponseSchema = z.object({
-  id: z.string(),
-});
+export const patientDtoSchema = demographicsSchema.merge(
+  z.object({
+    id: createNonEmptryStringSchema("id"),
+    externalId: z.string().nullish(),
```

Comment:
Same comment as `personalIdentifier` above -- I don't think we should have `null` here but undefined or string...